libavfilter/vf_lvpdnn.c

```diff
@@ -435,8 +348,170 @@ static av_cold void uninit(AVFilterContext *ctx)
 
     if(context->log_filename && context->logfile)
     {
-        fclose(context->logfile);        
+        fclose(context->logfile);
+    }
+}
+static int create_dnnmodel(LVPDnnLoadData** models, char *modelpath, char *input, char *output, int deviceid)
+{
+    int result = 0;
+    LVPDnnLoadData* dnndata;
+
+    if(models[deviceid]) {
+        av_log(NULL, AV_LOG_WARNING, "model data already created before\n");
+        return result;
+    }
+
+    dnndata = (LVPDnnLoadData*)av_mallocz(sizeof(LVPDnnLoadData));
+    if(!dnndata) {
+        av_log(NULL, AV_LOG_ERROR, "could not create LVPDnnLoadData buffer\n");
+        return AVERROR(EIO);
+    }
+
+    models[deviceid] = dnndata;
+    //copy parameters, will use to compare filter argument
+    dnndata->model_filename = (char*)av_mallocz(MAX_STRING_SIZE);
+    dnndata->model_inputname = (char*)av_mallocz(MAX_STRING_SIZE);
+    dnndata->model_outputname = (char*)av_mallocz(MAX_STRING_SIZE);
+    if (!dnndata->model_filename || !dnndata->model_inputname || !dnndata->model_outputname) {
+        av_log(NULL, AV_LOG_ERROR, "could not create parameter string\n");
+        return AVERROR(ENOMEM);
+    }
+
+    strcpy(dnndata->model_filename, modelpath);
+    strcpy(dnndata->model_inputname, input);
+    strcpy(dnndata->model_outputname, output);
+    //init dnn model
+    dnndata->backend_type = DNN_TF;
+    dnndata->dnn_module = ff_get_dnn_module(dnndata->backend_type);
+    if (!dnndata->dnn_module) {
+        av_log(NULL, AV_LOG_ERROR, "could not create DNN module for requested backend\n");
+        return AVERROR(ENOMEM);
+    }
+    //setting device id for model loading, 0x30 is tensorflow protobuf value for gpuid 0
```

Comment:
Is 0x30 guaranteed to always be GPU ID 0? Is there no way to avoid referencing protobuf values in this manner and its not possible to just do set_deviceid(deviceid) where deviceid is a decimal number?

Reply:
I guess I couldn't ignore protobuf. modified as the direct call manner.